File: Others/OctaveR/numI.r
integral = function(a, b) {
	acc <- 0.0001
	n = strtoi((b-a)/acc)
	delta_x = (b-a)/n
	s <- 0
	for(i in 1:n){
		x_i <- a + (i-0.5)*delta_x
		s <- s + f(x_i)*delta_x
	}
    return(s)
}

f = function(x){
	return(sin(x))
}

main = function(){
	a <- 0
	b <- 2
	a = as.numeric(a); b = as.numeric(b)
	ans <- integral(a, b)
	print('I[f(x)] with')
	print(paste('a = ', a))
	print(paste('b = ', b))
	print(paste('is equal to: ', ans))
}

if(getOption("numI.main", default=TRUE)){
   main()
}

File: Others/numI.jl
const acc = 0.0001

f(x) = sin(x)

function integral(a::Float64, b::Float64)
    global acc
    n = trunc(Int, (b-a)/acc)
    delta_x = (b-a)/n
    s = 0
    for i in 1:n
        x_i = a + (i-0.5)*delta_x
        s += f(x_i)*delta_x
    end
    return s
end

function main()
    print("Set a: ")
    a = parse(UInt8, readline())
    print("Set b: ")
    b = parse(UInt8, readline())
    ans = integral(a, b)
    println("I[sin(x)] with")
    println("a = $a")
    println("b = $b")
    println("is equal to: $ans")
end

main()